diff --git a/av2/encoder/speed_features.c b/av2/encoder/speed_features.c
index 48cc7b0f749c6d19658a2a50e90b9f58b24131fc..92e99a2fa2282954e4fe854eb23e7d4bc6bcd147 100644
--- a/av2/encoder/speed_features.c
+++ b/av2/encoder/speed_features.c
@@ -547,6 +547,9 @@ static void set_good_speed_features_framesize_independent(
     // TODO(any): Re-enable for inter frames after fixing speed feature.
     sf->winner_mode_sf.enable_winner_mode_for_tx_size_srch = 0;
     sf->winner_mode_sf.enable_winner_mode_for_use_tx_domain_dist = 1;
+    // Use the conservative DC-block predictor at speed 3. Speed 4 upgrades
+    // this to level 2 below, which also enables prediction in winner mode.
+    sf->winner_mode_sf.dc_blk_pred_level = 1;
     sf->winner_mode_sf.motion_mode_for_winner_cand =
         boosted                                                      ? 0
         : gf_group->update_type[gf_group->index] == INTNL_ARF_UPDATE ? 1
