diff --git a/av2/encoder/intra_mode_search.c b/av2/encoder/intra_mode_search.c
index 20e7a000be60947dd8014f23537772d82eeefd57..038d862c17d1127dabc275575e1ed6cef331b14e 100644
--- a/av2/encoder/intra_mode_search.c
+++ b/av2/encoder/intra_mode_search.c
@@ -1242,8 +1242,9 @@ int64_t av2_handle_intra_mode(IntraModeSearchState *intra_search_state,
   }
 
   int64_t this_model_rd = intra_model_yrd(cpi, x, bsize, mode_cost);
-  const int k =
-      cpi->sf.intra_sf.intra_pruning_with_mlp ? 4 : TOP_INTRA_MODEL_COUNT;
+  const int k = cpi->sf.intra_sf.intra_pruning_with_mlp
+                    ? (cpi->oxcf.speed >= 4 ? 2 : 4)
+                    : TOP_INTRA_MODEL_COUNT;
   if (prune_intra_y_mode(this_model_rd, best_model_rd, top_intra_model_rd, k,
                          xd->lossless[mbmi->segment_id], mbmi->use_dpcm_y))
     return INT64_MAX;
@@ -1533,7 +1534,9 @@ void search_fsc_mode(const AV2_COMP *const cpi, MACROBLOCK *x, int *rate,
           this_model_rd = intra_model_yrd(cpi, x, bsize, mode_costs);
 
           if (prune_intra_y_mode(this_model_rd, best_model_rd,
-                                 top_intra_model_rd, TOP_INTRA_MODEL_COUNT,
+                                 top_intra_model_rd,
+                                 cpi->oxcf.speed >= 4 ? 2
+                                                      : TOP_INTRA_MODEL_COUNT,
                                  xd->lossless[mbmi->segment_id],
                                  mbmi->use_dpcm_y)) {
             continue;
@@ -1738,8 +1741,9 @@ int64_t av2_rd_pick_intra_sby_mode(const AV2_COMP *const cpi, ThreadData *td,
   mbmi->dpcm_mode_y = 0;
   // mbmi->dpcm_angle_delta = 0;
   //  Searches the intra-modes except for intrabc, palette, and filter_intra.
-  const int model_rd_k =
-      cpi->sf.intra_sf.intra_pruning_with_mlp ? 4 : TOP_INTRA_MODEL_COUNT;
+  const int model_rd_k = cpi->sf.intra_sf.intra_pruning_with_mlp
+                             ? (cpi->oxcf.speed >= 4 ? 2 : 4)
+                             : TOP_INTRA_MODEL_COUNT;
   int64_t top_intra_model_rd[TOP_INTRA_MODEL_COUNT];
   for (int i = 0; i < TOP_INTRA_MODEL_COUNT; i++) {
     top_intra_model_rd[i] = INT64_MAX;
